win32/winapp.cpp

```diff
@@ -693,50 +694,48 @@ LRESULT WINAPI MyWndProc(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam) {
     break;
   }
 
-  oeWin32Application *winapp = Win32_AppObjects[i].app;
-
   //	if this window not on list, then run default window proc.
-  if (i == -1 || winapp == NULL || force_default)
+  if (i < 0 || Win32_AppObjects[i].app == NULL || force_default)
     return DefWindowProc(hWnd, msg, wParam, lParam);
 
-  if (!winapp->run_handler((HWnd)hWnd, (unsigned)msg, (WParam)wParam, (LParam)lParam))
+  if (!Win32_AppObjects[i].app->run_handler((HWnd)hWnd, (unsigned)msg, (WParam)wParam, (LParam)lParam))
     return 0;
 
   // run user defined window procedure.
-  return (LRESULT)winapp->WndProc((HWnd)hWnd, (unsigned)msg, (WParam)wParam, (LParam)lParam);
+  return (LRESULT)Win32_AppObjects[i].app->WndProc((HWnd)hWnd, (unsigned)msg, (WParam)wParam, (LParam)lParam);
 }
 
 // detect if application can handle what we want of it.
 bool oeWin32Application::GetSystemSpecs(const char *fname) {
   FILE *fp = fopen(fname, "wt");
   tWin32OS os;
   int maj, min, build;
-  char desc[256];
+  char desc[256]{};
 
   if (!fp)
     return false;
 
   os = oeWin32Application::version(&maj, &min, &build, desc);
 
-  fprintf(0, "OS: %s %d.%d.%d %s\n",
+  fprintf(fp, "OS: %s %d.%d.%d %s\n",
           (os == Win9x)   ? "Win9x"
           : (os == WinNT) ? "WinNT"
           : (os == WinCE) ? "WinCE"
                           : "Non standard Win32",
           maj, min, build, desc);
 
   // get system memory info
-  MEMORYSTATUS mem_stat;
+  MEMORYSTATUSEX mem_stat{};
```

Comment:
So what's the difference between the 2 APIs ? are they ABI compatible?

Reply:
> On computers with more than 4 GB of memory, the GlobalMemoryStatus function can return incorrect information, reporting a value of –1 to indicate an overflow. For this reason, applications should use the [GlobalMemoryStatusEx](https://learn.microsoft.com/en-us/windows/desktop/api/sysinfoapi/nf-sysinfoapi-globalmemorystatusex) function instead.

`GlobalMemoryStatus` is a different method endpoint from `GlobalMemoryStatusEx`, so no ABI worries passing the different structure type.

This _does_ come with a minimum supported version of Windows XP, however. I don't know if Win95 support is something we want to maintain.

Reply:
Win95 does not matter to us, and I'm pretty sure we're already incompatible, we can safely ignore that